npm/vite-dev-server/src/plugins/cypress.ts

```diff
@@ -52,21 +53,45 @@ export const Cypress = (
     configResolved (config) {
       base = config.base
     },
-    async transformIndexHtml () {
+    async transformIndexHtml (html) {
+      // it's possibe other plugins have modified the HTML
+      // before we get to. For example vitejs/plugin-react will
+      // add a preamble. We do our best to look at the HTML we
+      // receive and inject it.
+      // For now we just grab any `<script>` tags and inject them to <head>.
+      // We will add more handling as we learn the use cases.
+      const root = parse(html)
+
+      const scriptTagsToInject = root.childNodes.filter((node) => {
+        // @ts-ignore
+        return node.rawTagName === 'script'
```

Comment:
Also, the type defs in `node-html-parser` seem to be wrong.

Reply:
I think this is because `childNodes` is an array of `Node` which is an abstract class - implementations include `TextNode` for comments (which have no tag name) and `HTMLElement` which has the `rawTagName` field. To make TS happy you'd need to do a type-check here then you could use the field name without the linter override

Reply:
Nice, I just did:

```js
const scriptTagsToInject = root.childNodes.filter((node) => {
  return node instanceof HTMLElement && node.rawTagName === 'script'
})
```

Reply:
Here: https://github.com/cypress-io/cypress/pull/22915/commits/0d12d60b24b720fa62e41a5687ce7d62a604e225